sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -367,6 +372,551 @@ block_store(Tx *addr, simd<Tx, N> vals, Flags = {}) {
+  check_lsc_vector_size<FactoredNElts>();
+
+  // Prepare template arguments for the call of intrinsic.
+  constexpr lsc_data_size ActualDS =
+      Use64BitData ? lsc_data_size::u64 : lsc_data_size::u32;
+
+  constexpr lsc_vector_size _VS = to_lsc_vector_size<FactoredNElts>();
+  using LoadElemT = __ESIMD_DNS::__raw_t<
+      std::conditional_t<SmallIntFactor == 1, T,
+                         std::conditional_t<Use64BitData, uint64_t, uint32_t>>>;
+
+  constexpr uint16_t _AddressScale = 1;
+  constexpr int _ImmOffset = 0;
+
+  constexpr auto _Transposed = lsc_data_order::transpose;
+  constexpr int N = 1;
+
+  simd<uintptr_t, N> Addrs = reinterpret_cast<uintptr_t>(p);
+  simd<LoadElemT, FactoredNElts> OldVals =
+      old_values.template bit_cast_view<LoadElemT>();
+  simd<LoadElemT, FactoredNElts> Result =
+      __esimd_lsc_load_merge_stateless<LoadElemT, L1H, L2H, _AddressScale,
+                                       _ImmOffset, ActualDS, _VS, _Transposed,
+                                       N>(pred.data(), Addrs.data(),
+                                          OldVals.data());
+  return Result.template bit_cast_view<T>();
+}
+
+} // namespace detail
+
+/// @addtogroup sycl_esimd_memory_block
+/// @{
+
+/// Each of the following block load functions loads a contiguous memory block
+/// from the address referenced by the USM pointer 'ptr', or from 'ptr +
+/// offset', where 'offset' is the offset in bytes (not in elements!). The
+/// parameter 'pred' is the one element predicate. If it is set to 1, then all
+/// 'N' elements are loaded. Otherwise, the block load operation is a NO-OP. The
+/// parameter 'old_values' specifies the values being copied to the returned
+/// result if 'pred' is set to 0.
+/// The parameter 'props' specifies the optional compile-time properties
+/// of the type esimd::properties and may include esimd::cache_hint_L1,
+/// esimd::cache_hint_L2, esimd::cache_hint_L3, esimd::alignment.
+
+/// simd<T, N> block_load(const T* ptr, props={});                      // (1)
+/// simd<T, N> block_load(const T* ptr, size_t offset, props={});       // (2)
+
+/// simd<T, N> block_load(const T* ptr, simd_mask<1> pred, props={});   // (3)
+/// simd<T, N> block_load(const T* ptr, size_t offset, simd_mask<1> pred,
+///                       props={});                                    // (4)
+
+/// simd<T, N> block_load(const T* ptr, simd_mask<1> pred,
+///                       simd<T, N> old_values, props={});             // (5)
+/// simd<T, N> block_load(const T* ptr, size_t offset, simd_mask<1> pred,
+///                       simd<T, N> old_values, props={});             // (6)
+
+/// simd<T, N> block_load(const T* ptr, props={}); // (1)
+/// This function loads a contiguous memory block from USM pointer \p ptr.
+///
+/// It may have temporary restrictions depending on L1, L2 cache hints,
+/// See details at 'Restrictions' section below. The restrictions will be
+/// relaxed eventually/soon.
```

Comment:
```suggestion
/// relaxed in the future.
```

Maybe we don't commit to anything :)

Reply:
Fixed in c8a7f8388af16204b1ced6d2d265f6e341944386